tools/deviceGen/input/device_limits.json

```diff
@@ -0,0 +1,2347 @@
+            "value": false,
+            "comment": "TODO: Actually implement the sampler flag enums"
+        },
+        {
+            "type": "bool",
+            "name": "vulkanMemoryModelAvailabilityVisibilityChains",
+            "value": false,
+            "comment": "Vulkan 1.3 requires but we make concessions for MoltenVK"
+        },
+        {
+            "groupComment": [
+                "Vulkan 1.2 Core or VK_EXT_shader_viewport_index_layer"
+            ]
+        },
+        {
+            "type": "bool",
+            "name": "shaderOutputViewportIndex",
+            "value": false,
+            "comment": "ALIAS: VK_EXT_shader_viewport_index_layer"
+        },
+        {
+            "type": "bool",
+            "name": "shaderOutputLayer",
+            "value": false,
+            "comment": "ALIAS: VK_EXT_shader_viewport_index_layer"
+        },
+        {
+            "groupComment": [
+                "Vulkan 1.3 non-optional requires but poor support"
+            ]
+        },
+        {
+            "type": "bool",
+            "name": "shaderDemoteToHelperInvocation",
+            "value": false,
+            "comment": "or VK_EXT_shader_demote_to_helper_invocation"
+        },
+        {
+            "type": "bool",
+            "name": "shaderTerminateInvocation",
+            "value": false,
+            "comment": "or VK_KHR_shader_terminate_invocation"
+        },
+        {
+            "groupComment": [
+                "Vulkan 1.3 non-optional requires but poor support"
+            ]
+        },
+        {
+            "type": "bool",
+            "name": "shaderZeroInitializeWorkgroupMemory",
+            "value": false,
+            "comment": "or VK_KHR_zero_initialize_workgroup_memory"
+        }
+    ],
+    "nabla": [
+        {
+            "groupComment": [
+                "Nabla"
+            ]
+        },
+        {
+            "type": "uint32_t",
+            "name": "computeUnits",
+            "value": 0
+        },
+        {
+            "type": "bool",
+            "name": "dispatchBase",
+            "value": false,
+            "comment": "true in Vk, false in GL"
+        },
+        {
+            "type": "bool",
+            "name": "allowCommandBufferQueryCopies",
+            "value": false
+        },
+        {
+            "type": "uint32_t",
+            "name": "maxOptimallyResidentWorkgroupInvocations",
+            "value": 0,
+            "comment": "its 1D because multidimensional workgroups are an illusion"
+        },
+        {
+            "type": "uint32_t",
+            "name": "maxResidentInvocations",
+            "value": 0,
+            "comment": "These are maximum number of invocations you could expect to execute simultaneously on this device"
+        },
+        {
+            "type": "asset::CGLSLCompiler::E_SPIRV_VERSION",
+            "name": "spirvVersion",
+            "value": "asset::CGLSLCompiler::E_SPIRV_VERSION::ESV_1_6"
```

Comment:
we need to move the definition of the enum from `asset::CGLSLCompiler` to `nbl::hlsl`